src/SFA.DAS.EmployerApprenticeshipsService.Application/Queries/GetActivities/GetActivitiesQueryHandler.cs

```diff
@@ -0,0 +1,78 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using MediatR;
+using SFA.DAS.Activities;
+using SFA.DAS.Activities.Client;
+using SFA.DAS.EAS.Domain.Data.Repositories;
+using SFA.DAS.EAS.Domain.Models.UserProfile;
+using SFA.DAS.NLog.Logger;
+
+namespace SFA.DAS.EAS.Application.Queries.GetActivities
+{
+    public class GetActivitiesQueryHandler : IAsyncRequestHandler<GetActivitiesQuery, GetActivitiesResponse>
+    {
+        private readonly CurrentUser _currentUser;
+        private readonly IActivitiesClient _activitiesClient;
+        private readonly ILog _logger;
+        private readonly IMembershipRepository _membershipRepository;
+
+        public GetActivitiesQueryHandler(
+            CurrentUser currentUser,
+            IActivitiesClient activitiesClient,
+            ILog logger,
+            IMembershipRepository membershipRepository)
+        {
+            _currentUser = currentUser;
+            _activitiesClient = activitiesClient;
+            _logger = logger;
+            _membershipRepository = membershipRepository;
+        }
+
+        public async Task<GetActivitiesResponse> Handle(GetActivitiesQuery message)
+        {
+            var membership = await _membershipRepository.GetCaller(message.HashedAccountId, _currentUser.ExternalUserId);
```

Comment:
I'm wondering if having a null/empty check on both hashedAccountId and ExternalUserId would be useful here just in case this handler is called from code that is not directly linked to and actions request?

Reply:
Nice catch, thanks. I'll add a handler into the Mediatr pipeline at some point that does it.